mne/time_frequency/tfr.py

```diff
@@ -1182,9 +1190,12 @@ def tfr_morlet(inst, freqs, n_cycles, use_fft=False,
     tfr_multitaper, tfr_stockwell
     """
     data = _get_data(inst, return_itc)
-    picks = pick_types(inst.info, meg=True, eeg=True)
-    info = pick_info(inst.info, picks)
-    data = data[:, picks, :]
+    info = inst.info
+
+    info, picks = _prepare_picks_info(info, picks)
+    if not np.array_equal(picks, np.arange(len(data))):
```

Comment:
instead of adding multiple versions of these conditionals, you can put a conditional in `_prepare_picks_info` that does the comparison and if it passes, sets `picks = slice(None)`. That way here and above you won't need a conditional at all as `data = data[:, picks, :]` will do nothing.